packages/widget-chat/src/components/activity-item/index.js

```diff
@@ -1,36 +1,42 @@
 import React, {PropTypes} from 'react';
 import classNames from 'classnames';
 import ActivityItemPost from '../activity-item-post';
+import ActivityItemShareList from '../activity-item-share-list';
 import ActivityItemSystemMessage, {SYSTEM_MESSAGE_VERBS} from '../activity-item-system-message';
 
 import styles from './styles.css';
 
 const POST_VERB = `post`;
+const SHARE_VERB = `share`;
 
 export default function ActivityItem(props) {
   const {
+    activity,
     verb
   } = props;
 
   let itemComponent = ``;
 
   if (verb === POST_VERB) {
-    itemComponent = <ActivityItemPost {...props} />;
+    itemComponent = <ActivityItemPost content={activity.displayName} {...props} />;
   }
   else if (SYSTEM_MESSAGE_VERBS.indexOf(verb) !== -1) {
-    itemComponent = <ActivityItemSystemMessage {...props} />;
+    itemComponent = <ActivityItemSystemMessage content={activity.displayName} {...props} />;
+  }
+  else if (verb === SHARE_VERB) {
```

Comment:
Yea, that way, we only do an array search as a last resort (possibly saving up to 5 cpu cycles a year! 😆 )